<?xml version="1.0" ?>
<launch>
  <!--Be sure to pass the IPs of your pandas like robot_ips:={panda_1/robot_ip: <my_ip_1>, panda_2/robot_ip: <my_ip_2>}   -->
  <arg name="robot_ips" />
  <arg name="panda_right"/>
  <arg name="panda_left"/>
  <arg name="robot_id" default="panda_dual" />
  <arg name="robot_id_right" default="panda_right" />
  <arg name="robot_id_left" default="panda_left" />
  <arg name="rviz" default="true" />
  <arg name="rqt" default="true" />

  <include file="$(find franka_bimanual_controllers)/launch/franka_combined_control.launch" >
    <arg name="robot_id" value="$(arg robot_id)" />
    <arg name="robot_ips" value="$(arg robot_ips)" />
  </include>

  <group ns="$(arg robot_id)">
    <rosparam command="load" file="$(find franka_bimanual_controllers)/config/franka_bimanual_controllers.yaml" />
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"  args="bimanual_cartesian_impedance_controller"/>
    <node pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find franka_bimanual_controllers)/launch/rviz/franka_dual_description_with_marker.rviz" if="$(arg rviz)"/>
  </group>
   <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" required="false" /> 

  <arg name="robot_ip" />
  <arg name="stop_at_shutdown" default="false" />
  <arg name="arm_id"      default="panda" />
  <arg name="joint_names" default="[$(arg arm_id)_finger_joint1, $(arg arm_id)_finger_joint2]" />

  <node name="$(arg robot_id_right)_franka_gripper" pkg="franka_gripper" type="franka_gripper_node" output="screen">
    <param name="robot_ip" value="$(arg panda_right)"/>
    <param name="stop_at_shutdown" value="$(arg stop_at_shutdown)" />
    <rosparam command="load" file="$(find franka_gripper)/config/franka_gripper_node.yaml" />
    <rosparam param="joint_names" subst_value="true">$(arg joint_names)</rosparam>
  </node>

  <node name="$(arg robot_id_left)_franka_gripper" pkg="franka_gripper" type="franka_gripper_node" output="screen">
    <param name="robot_ip" value="$(arg panda_left)"/>
    <param name="stop_at_shutdown" value="$(arg stop_at_shutdown)" />
    <rosparam command="load" file="$(find franka_gripper)/config/franka_gripper_node.yaml" />
    <rosparam param="joint_names" subst_value="true">$(arg joint_names)</rosparam>
  </node>
</launch>
